contracts/erc20-bridge-sampler/contracts/src/ERC20BridgeSampler.sol

```diff
@@ -41,6 +41,61 @@ contract ERC20BridgeSampler is
     uint256 constant internal UNISWAP_SAMPLE_CALL_GAS = 150e3;
     uint256 constant internal ETH2DAI_SAMPLE_CALL_GAS = 250e3;
 
+     struct OrdersAndSample {
+        uint256[] orderFillableTakerAssetAmounts;
+        uint256[][] makerTokenAmountsBySource;
+    }
+
+    function queryMultipleOrdersAndSampleBuys(
```

Comment:
I think it's probably clearer to rename these functions to something just like `getTokenBuy/SellPrices()`, and return prices instead of fill amounts. For every pair, we could just return the price of the first fillable native order + the price from each DEX. Price could be expressed either as an 18 decimal fraction or a maker/taker amount pair.

As is, it's kind of hard to understand what the intent of these functions are.